resources/shuttle-openai.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: "Integrate OpenAI with Shuttle"
+icon: "microchip"
+---
+
+# Integrate OpenAI with Shuttle
+
+This plugin allows services to connect to [OpenAI](https://openai.com/), enabling easy access to OpenAI's powerful language models and AI capabilities.
+
+## Usage
+
+Add `shuttle-openai` to the dependencies for your service by running `cargo add shuttle-openai`.
+
+This resource will be provided by adding the `shuttle_openai::OpenAI` attribute to your Shuttle `main` decorated function.
+
+It returns an `async_openai::Client<OpenAIConfig>` for you to interact with OpenAI's services.
+
+### Example
+
+In the case of an Axum server, your main function will look like this:
+
+```rust
+use async_openai::{Client, config::OpenAIConfig};
+use shuttle_axum::ShuttleAxum;
+
+#[shuttle_runtime::main]
+async fn app(
+    #[shuttle_openai::OpenAI(api_key = "{secrets.OPENAI_API_KEY}")]
+    openai: Client<OpenAIConfig>,
+) -> ShuttleAxum {
+    // Your app logic here
+}
+```
+
+[Click here for the full example.](https://github.com/shuttle-hq/shuttle-examples/tree/main/axum/openai)
+
+### Parameters
+
+| Parameter | Type  | Description                          |
+|-----------|-------|--------------------------------------|
+| api_key   | `str` | The API key for OpenAI authentication |
```

Comment:
There are 3 other optional strings that we should list here. Check https://github.com/shuttle-hq/shuttle/commit/e2e178cf0b98c75a005382fc29535abaebe771ba#diff-66f7ebd13c63a9e4a906fd7d1dd8fc1a8b7864e3517439afde2a5bc70ac9e18cR38-R41

Reply:
Done, thanks 👍